modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/Reader.java

```diff
@@ -280,28 +280,12 @@ protected void processParameter(List<Parameter> parameters, Operation operation,
                 if (operation.getRequestBody() == null) {
                     io.swagger.oas.annotations.parameters.RequestBody requestBodyAnnotation = getRequestBody(Arrays.asList(paramAnnotations));
                     if (requestBodyAnnotation != null) {
-                        Optional<RequestBody> optionalRequestBody = OperationParser.getRequestBody(requestBodyAnnotation, components);
+                        Optional<RequestBody> optionalRequestBody = OperationParser.getRequestBody(requestBodyAnnotation, classConsumes, methodConsumes, components);
                         if (optionalRequestBody.isPresent()) {
                             requestBody = optionalRequestBody.get();
                             if (StringUtils.isBlank(requestBody.get$ref()) &&
                                     (requestBody.getContent() == null || requestBody.getContent().isEmpty())) {
                                 if (parameter.getSchema() != null) {
-                                    Content content = new Content();
-                                    if (methodConsumes != null) {
-                                        for (String value : methodConsumes.value()) {
-                                            setMediaTypeToContent(parameter.getSchema(), content, value);
-                                        }
-                                    } else if (classConsumes != null) {
-                                        for (String value : classConsumes.value()) {
-                                            setMediaTypeToContent(parameter.getSchema(), content, value);
-                                        }
-                                    } else {
-                                        setMediaTypeToContent(parameter.getSchema(), content, DEFAULT_MEDIA_TYPE_VALUE);
-                                    }
-
-                                    requestBody.setContent(content);
-                                }
-                                if (parameter.getSchema() != null) {
```

Comment:
Why has the above been removed? its purpose was populating request body schema/content from parameter schema if not set in annotation.

Reply:
This is now treated inside the getContent method, the consumes and produces are now treated as String[] mediaTypes.

Reply:
@ralphdoe Can you point me to code implementing the above? getContent doesn't seem to use parameter schema if not set in annotation, but maybe I am wrong

Reply:
This is the getRequestBody method in OperationParser: https://github.com/swagger-api/swagger-core/blob/bugfx/2395-reader-annotations-improvent/modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java#L341-L360. Also you can see: https://github.com/swagger-api/swagger-core/blob/bugfx/2395-reader-annotations-improvent/modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java#L452-L465

Reply:
@ralphdoe I might be wrong, but that code seems to populate requestBody with values from @RequestBody and @Consumes annotations, which is not what the removed code did, which was applying the so far resolved parameter schema to requestBody if requestBody doesn't specify it

Reply:
My bad, code recovered.